File: CODE_FILES/TASK1/PYTHON/compress.py
import gzip
import time

start = time.time()

with open("input.txt", "rb") as f_in:
    with gzip.open("outputpy.txt.gz", "wb") as f_out:
        f_out.writelines(f_in)

end = time.time()
print(f"Time taken: {end - start:.4f} seconds")

File: CODE_FILES/TASK1/GO/compress.go
package main

import (
    "compress/gzip"
    "io"
    "os"
    "time"
)

func main() {
    start := time.Now()

    inputFile, _ := os.Open("input.txt")
    defer inputFile.Close()

    outputFile, _ := os.Create("outputgo.txt.gz")
    defer outputFile.Close()

    writer := gzip.NewWriter(outputFile)
    defer writer.Close()

    io.Copy(writer, inputFile)

    elapsed := time.Since(start)
    println("Time taken:", elapsed.Milliseconds(), "ms")
}
